### PR TITLE
[exporter/awsxray] Fix X-Ray Segment status code 

### PR DESCRIPTION
**Pull request by mxiamxia**
**Description:** <Describe what has changed.>
The conversion from HTTP spans to X-Ray segments did not accurately capture the HTTP status code during the translation of the cause. As a result, some failed HTTP requests with 4xx and 5xx status codes were displayed as having an "OK" status in the X-Ray console. Also, add remote attribute in Exception for X-Ray remote segments.

**Link to tracking Issue:** <Issue number if applicable>
N/A

**Testing:** <Describe what testing was performed and which tests were added.>
unit test
**Documentation:** <Describe the documentation added.>
N/A

### CODE REVIEW

File: exporter/awsxrayexporter/internal/translator/cause.go [hunk 1]
```diff
@@ -130,11 +128,19 @@ func makeCause(span ptrace.Span, attributes map[string]pcommon.Value, resource p
 			if code == 429 {
 				isThrottle = true
 			}
-		} else {
+		} else if code >= 500 && code <= 599 {
 			isError = false
 			isThrottle = false
 			isFault = true
+		} else {
```
Comment (bjrara) [inline exporter/awsxrayexporter/internal/translator/cause.go:135]: Should we check span status code before we set `isError=false`?

Comment (mxiamxia) [reply exporter/awsxrayexporter/internal/translator/cause.go:135]: I was thinking to use `http.status_code` alone to determine X-Ray segment status, but your suggestion make sense for the edge cases. We'll still check status_code `400-599` for determine the error/fault segments first, and then use Span status.code() to set `isFault` in X-Ray segment if the status code is true

**Comment by bjrara** [comment]
LGTM

**Review by bjrara — APPROVED** [review]

**Review by wangzlei — APPROVED** [review]

**Comment by mxiamxia** [comment]
Hi @dashpole, could you please take a look this PR when you get a chance

**Review by srprash — APPROVED** [review]
LGTM with some nits.

**Comment by mxiamxia** [comment]
Hi @dashpole , kindly follow if you could help to take a look this PR. Ty!

**Comment by mxiamxia** [comment]
Hi can I get some help from @open-telemetry/collector-contrib-approvers to review this PR?

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**